packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -727,11 +797,33 @@ class GradleForAgp {
     required this.agpMax,
     required this.minRequiredGradle,
   });
+
   final String agpMin;
   final String agpMax;
   final String minRequiredGradle;
 }
 
+// Data class that represents a range of versions.
+@immutable
+class VersionRange{
```

Comment:
Looking for feedback on using this as a convenience class, because I realize it overlaps with the other classes defined here and could also be tied more closely to the `Version` class.

Reply:
Can you put this in its own file if you are going to make a utility class?